schematics/types/base.py

```diff
@@ -429,14 +438,14 @@ def _mock(self, context=None):
     def to_native(self, value, context=None):
         if isinstance(value, bool):
             value = int(value)
-        if isinstance(value, self.number_class):
+        if isinstance(value, self.primitive_type):
```

Comment:
`native_type` and `primitive_type` are the same, but I agree it would be more intuitive to use `primitive_type` within `to_primitive`.  If backward compatibility is a concern I could also keep `number_class`.